editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/maven/CamelProjectConfigurator.java

```diff
@@ -136,6 +139,14 @@ private void configureFacet(MavenProject mavenProject, IProject project, IProgre
 		if (!isCamelConfigurable(mavenProject, project)) {
 			return;
 		}
+		
+		// pause to make sure all builds are done
+        try {
+            waitJob();
+        } catch (OperationCanceledException | InterruptedException e) {
+            ProjectTemplatesActivator.pluginLog().logError(e);
+            return;
+        }
 		IFacetedProject fproj = ProjectFacetsManager.create(project);
```

Comment:
For some reason it appears to be dropping through this code completely unless you debug into it, which gives whatever thread is still processing enough time to finish. But the facet nature itself is never added to the project, so none of the rest of it happens. I thought that by waiting for builds to finish it might clear it up, but as you can see that's not the case.